governance/program/tests/program_test/mod.rs

```diff
@@ -3143,4 +3143,42 @@ impl GovernanceProgramTest {
 
         Ok(max_voter_weight_record_cookie)
     }
+
+    #[allow(dead_code)]
+    pub async fn with_complete_proposal(
+        &mut self,
+        governance_cookie: &mut GovernanceCookie,
+        proposal_cookie: &mut ProposalCookie,
+    ) -> Result<(), ProgramError> {
+        self.with_complete_proposal_using_instruction_impl(
+            governance_cookie,
+            proposal_cookie,
+            NopOverride,
+        )
+        .await
+    }
+
+    #[allow(dead_code)]
+    pub async fn with_complete_proposal_using_instruction_impl<F: Fn(&mut Instruction)>(
+        &mut self,
+        governance_cookie: &mut GovernanceCookie,
+        proposal_cookie: &mut ProposalCookie,
+        instruction_override: F,
+    ) -> Result<(), ProgramError> {
+        let mut complete_proposal_ix = complete_proposal(
+            &self.program_id,
+            &governance_cookie.address,
+            &proposal_cookie.address,
+        );
+        instruction_override(&mut complete_proposal_ix);
+
+        self.bench
+            .process_transaction(&[complete_proposal_ix], None)
+            .await?;
+
+        let proposal_data = self.get_proposal_account(&proposal_cookie.address).await;
```

Comment:
Do not use actual state in the test sdk functions. The idea is that they should return what is expected and then you can compare, and assert, with the actual state in the test